src/spec.c

```diff
@@ -146,6 +146,9 @@ static int parseFieldSpec(const char **argv, int *offset, int argc, FieldSpec *s
   sp->sortIdx = -1;
   sp->options = 0;
   // the field name comes here
+  if (sp->name) {
+    free(sp->name);
```

Comment:
why do you need to free the name here?

Reply:
In case the result was from a prior failed allocation via ALTER. Maybe I can do it a different way?